lib/rubocop/cop/style/trailing_body_on_method_definition.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for trailing code after the method definition.
+      #
+      # @example
+      #   # bad
+      #   def some_method; do_stuff
+      #   end
+      #
+      #   # good
+      #   def some_method
+      #     do_stuff
+      #   end
+      #
+      class TrailingBodyOnMethodDefinition < Cop
+        include AutocorrectAlignment
+
+        MSG = 'Method body goes below definition.'.freeze
+
+        def on_def(node)
+          return unless node.body
+          return unless trailing_body?(node)
+
+          add_offense(node)
+        end
+        alias on_defs on_def
+
+        private
+
+        def trailing_body?(node)
+          node.line_count >= 2 && body_start_on_first_line?(node)
+        end
+
+        def body_start_on_first_line?(node)
+          node.source_range.first_line == node.body.source_range.first_line
+        end
+
+        def autocorrect(node)
+          body = node.body
```

Comment:
I don't think this needs a local variable, as it doesn't add anything in terms of readability, and is only used once. 🙂